linode/linode_instance_helpers.go

```diff
@@ -692,8 +692,18 @@ func createRandomRootPassword() (string, error) {
 
 // changeInstanceType resizes the Linode Instance
 func changeInstanceType(client *linodego.Client, instance *linodego.Instance, targetType string, d *schema.ResourceData) error {
+
+	diskResize := false
+	waitForOnline := true
+
+	resizeOpts := linodego.InstanceResizeOptions{
+		AllowAutoDiskResize: &diskResize,
+		Type:                targetType,
+	}
+
 	// Instance must be either offline or running (with no extra activity) to resize.
```

Comment:
Is it still possible (in the Linode API) to resize an instance while it is online or does it always have to be shutdown first?

If resizing requires an offline instance, it may be saner to always send a shutdown and await that status before sending a ResizeInstance (simplifying this code).

Reply:
We can issue a shutdown job to a Running instance... the resize job will shutdown the instance if necessary, and after the resize is complete, the Linode is returned to its original state (either Offline or Running).

If we always issued a shutdown job, we would still need to check the original status of the Linode (which we already do) and then return it to that state manually (which i think would add back the additional complexity we would save by always shutting down the instance)